Add a comment feature using the Comment class, doPost and doGet.

NaamaHolzer opened the pull request:
- Add a Comment class representing a user's comment
- Add doPost and doGet functions to support the comment feature
- Add a js function to deal with the JSON received from the servlet on get requests
- Add a comment form to the html file

NaamaHolzer committed:
Add a comment feature using the Comment class, doPost and doGet.

orarch commented on portfolio/pom.xml:
```diff
@@ -35,7 +40,7 @@
         <version>2.2.0</version>
         <configuration>
           <!-- TODO: set project ID. -->
```
The TODO comment can be removed

orarch commented on portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:
```diff
@@ -23,10 +27,59 @@
 
+  @Override
+  public void init() {
+    comments=new ArrayList<>();
```
nit: add spacing before and after assignment operator (comments = new ArrayList..)

Please apply to all assignments in this file

orarch commented on portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:
```diff
@@ -23,10 +27,59 @@
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
+//List of comments the portfolio received.
```
nit: Add spacing: "// List"

Please apply to all comments in this file

orarch commented on portfolio/src/main/webapp/script.js:
```diff
@@ -26,4 +26,34 @@ function addRandomFact() {
+}
+
+//Invoked when the pages is loaded. Gets the comments list from the server and displays it.
+ function getMessageFromServer(){
```
How about 'getCommentsFromServer'?

NaamaHolzer committed:
Update pom.xml

orarch commented on portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:
```diff
@@ -14,7 +14,11 @@
 
 package com.google.sps.servlets;
 
+import com.google.sps.data.Comment;
```
On what PR was this class created? I can't seem to find it.

orarch commented on portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:
```diff
@@ -23,10 +27,59 @@
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+      ArrayList<String> likedVal=new ArrayList<>();
+      //Creating a new Comment instance based on the new comment that was received.
+      String writerVal=getParameter(request,"writer","Anonymous");
```
I suggest replacing 'writer' with 'author'

orarch commented on portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:
```diff
@@ -23,10 +27,59 @@
+   * @return the request parameter, or the default value if the parameter
+   *         was not specified by the client
+   */
+  private String getParameter(HttpServletRequest request, String name, String defaultValue) {
```
This method can be static

orarch commented on portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:
```diff
@@ -23,10 +27,59 @@
+      //Add only checked checkboxes to the 'liked' list.
+      if(Boolean.parseBoolean(getParameter(request, "info", "false"))){
+          likedVal.add("The information");
+
```
formatting: remove empty line

orarch commented on portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:
```diff
@@ -23,10 +27,59 @@
+      //Creating a new Comment instance based on the new comment that was received.
+      String writerVal=getParameter(request,"writer","Anonymous");
+      //The received rate. Default value is 3.
+      int rangeVal=Integer.parseInt(getParameter(request, "rate", "3"));
```
Please be consistent, call it 'rateVal'

orarch commented on portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:
```diff
@@ -23,10 +27,59 @@
+      //The received rate. Default value is 3.
+      int rangeVal=Integer.parseInt(getParameter(request, "rate", "3"));
+      //Add only checked checkboxes to the 'liked' list.
+      if(Boolean.parseBoolean(getParameter(request, "info", "false"))){
```
For better readability I suggest renaming the "liked" attributes with some common pattern. 
For example: 'is_info_liked', 'is_facts_liked', etc.

Or, consider making this a param of list type

NaamaHolzer committed:
Add spacing before and after assignment operator

NaamaHolzer committed:
Add spacing to the comments

NaamaHolzer committed:
Change function name and add spacing

- Change function name from getMessageFromServer to GetCommentsFromServer
- Add spacing

NaamaHolzer committed:
Add the Comment class file

NaamaHolzer committed:
Add the Comment class file

NaamaHolzer committed:
Change 'writer' to 'author', make DataServlet's getParameter method static, and change variable names to be more consistent

orarch commented on portfolio/src/main/java/com/google/sps/data/Comment.java:
```diff
@@ -0,0 +1,40 @@
+    private ArrayList<String> liked;
+    private String text;
+    public Comment(String newAuthor, int newRate, ArrayList<String> newLiked, String newText){
+        this.liked = new ArrayList<>();
```
ArrayList has a copy constructer, so this can be replaced with: 

this.likedOptions = new ArrayList<>(newLikedOptions);

NaamaHolzer committed:
Use copy constructor for the likedOptions list in the Comment's constructor and change the name of the liked list from liked to likedOptions
